command/hook_count.go

```diff
@@ -90,6 +91,11 @@ func (h *CountHook) PostDiff(
 	h.Lock()
 	defer h.Unlock()
 
+	// We don't count enything for data sources
+	if strings.HasPrefix(n.Id, "data.") {
```

Comment:
Good question! I thought the same thing, but from reading [`HumanId()`](https://github.com/hashicorp/terraform/blob/edd67547bc2aa12244fcf1620bcb209ee73e008e/terraform/resource.go#L75-L84) I am fairly certain that `InstanceInfo` keeps `Id` unscoped and maintains `ModulePath []string` separately.

Will verify manually as well here.